backfila-embedded/src/main/kotlin/app/cash/backfila/embedded/BackfillRun.kt

```diff
@@ -32,9 +32,13 @@ interface BackfillRun<B : Backfill> {
 
   /** Prepares, scans and runs the whole backfill. */
   fun execute() {
-    precomputeRemaining()
-    scanRemaining()
-    runAllScanned()
+    try {
+      precomputeRemaining()
+      scanRemaining()
+      runAllScanned()
+    } finally {
+      finalizeBackfill()
```

Comment:
Yeah this is wrong. On error we shouldn't do this. Only on complete. MAYBE on marking the backfill cancelled.

Reply:
We will need a variable to mark when this is complete. Also this is meant to be how people interact with the backfill. People should never call this. It is an internal implementation detail. complete will be true if the finalize returned a success.